pkg/datapath/linux/routing/routing.go

```diff
@@ -192,6 +208,29 @@ func deleteRule(r route.Rule) error {
 	return errors.New("no rule found to delete")
 }
 
+// retrieveTableIDFromIfindex retrieves the rouitng table number / index from
+// the interface index. Interface here either means ENI or Azure interfaces.
+// For example, if the ifindex is 2 and that corresponds to "eth0", then we
+// will return 0 as the table index. If the ifindex corresponds to "eth4", then
+// we will return 4 as the table index.
+func retrieveTableIDFromIfindex(ifindex int) (int, error) {
+	iface, err := netlink.LinkByIndex(ifindex)
+	if err != nil {
+		return -1, fmt.Errorf("unable to get link by index: %w", err)
+	}
+
+	// We then use the interface name to get the integer part of the iface name
+	// (e.g. "0" from eth0, "1" from eth1).
+	idx, err := strconv.ParseInt(strings.ReplaceAll(iface.Attrs().Name, "eth", ""), 10, 64)
```

Comment:
Are Azure interfaces always "eth*"? I assumed this just to get something out to get feedback. This is a minor detail that can be iterated on later.

Reply:
I don't think the eth replace will work because the naming of the interfaces is different if the user enables or not (https://en.wikipedia.org/wiki/Consistent_Network_Device_Naming).

If this is disabled then I think the behavior is to keep `eth*` but if it's enabled it can differ (stock ubuntu AMI names interfaces `ens*`).

Reply:
Yeah, here's an example Ubuntu device I have on my laptop: `enp0s31f6`. These numbers are derived from the hardware layout of the system so that they would be consistent regardless of when each device is initialized during kernel boot.

Reply:
How can we reliably extract the "relevant" part from the ifname if don't know what the format is and is unpredictable?